### ISSUE TITLE
erigon stops synching after block number 0xe5a51a

### ISSUE THREAD

**Issue by TimDaub** [ledgerwatch/erigon#5184]
#### System information

./erigon/build/bin/erigon --version
erigon version 2021.11.3-alpha

OS & Version: 

uname -a
Linux ethnode 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

Commit hash : 0bb6478d3df64c130db407a783e5174deb78655f

#### Behaviour

- we have been running since a long time
- I usually check if the node is still synching by using the JSON rpc endpoint. And I interpret `result: false` as "the node is not synching because it is up to date with mainnet"

```
ps ax | grep erigon
  13392 ?        Sl   86165:09 ./erigon/build/bin/erigon --private.api.addr=localhost:9090 --datadir=/root/.local/share/erigon/
  14367 ?        Sl   1402:20 ./erigon/build/bin/rpcdaemon --private.api.addr=localhost:9090 --datadir=/root/.local/share/erigon/ --http.api=eth,web3,net,txpool
3696602 pts/1    S+     0:00 grep --color=auto erigon
```

as you can see the service is running and here are it's last logs with timestamps

```
tail node.out
[INFO] [08-25|15:22:30.244] [1/15 Headers] Wrote block headers       number=15050010 blk/second=0.000 alloc="4.84 GiB" sys="22.43 GiB"
[INFO] [08-25|15:22:51.883] [p2p] GoodPeers                          eth66=100
[INFO] [08-25|15:22:51.913] [txpool] stat                            block=15050010 pending=10000 baseFee=30000 queued=30000 alloc_mb=5055 sys_mb=22964
[INFO] [08-25|15:23:00.244] [1/15 Headers] Wrote block headers       number=15050010 blk/second=0.000 alloc="4.99 GiB" sys="22.43 GiB"
[INFO] [08-25|15:23:30.244] [1/15 Headers] Wrote block headers       number=15050010 blk/second=0.000 alloc="5.09 GiB" sys="22.43 GiB"
[INFO] [08-25|15:23:51.913] [txpool] stat                            block=15050010 pending=10000 baseFee=30000 queued=30000 alloc_mb=5315 sys_mb=22964
[INFO] [08-25|15:24:00.244] [1/15 Headers] Wrote block headers       number=15050010 blk/second=0.000 alloc="4.70 GiB" sys="22.43 GiB"
[INFO] [08-25|15:24:30.244] [1/15 Headers] Wrote block headers       number=15050010 blk/second=0.000 alloc="4.83 GiB" sys="22.43 GiB"
[INFO] [08-25|15:24:51.883] [p2p] GoodPeers                          eth66=100
[INFO] [08-25|15:24:51.913] [txpool] stat                            block=15050010 pending=10000 baseFee=30000 queued=30000 alloc_mb=5036 sys_mb=22964
```

however, what is odd is that synching is false but clearly the node is 56 days behind mainnet https://etherscan.io/block/15050010

```
root@ethnode ~ # curl -s -X POST  -H "Content-Type: application/json"   --data '{"jsonrpc": "2.0","method": "eth_blockNumber","params": [],"id": 0}'   http://localhost:8545
{"jsonrpc":"2.0","id":0,"result":"0xe5a51a"}
root@ethnode ~ # curl -s -X POST  -H "Content-Type: application/json"   --data '{"jsonrpc": "2.0","method": "eth_syncing","params": [],"id": 0}'   http://localhost:8545
{"jsonrpc":"2.0","id":0,"result":false}
```

```
df -h
Filesystem      Size  Used Avail Use% Mounted on
udev             63G     0   63G   0% /dev
tmpfs            13G  1.0M   13G   1% /run
/dev/nvme0n1p3  3.5T  2.7T  643G  81% /   <==== erigon disk
tmpfs            63G     0   63G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs            63G     0   63G   0% /sys/fs/cgroup
/dev/nvme0n1p2  975M   83M  841M   9% /boot
tmpfs            13G     0   13G   0% /run/user/0
```

#### Expected behaviour

- there is two things
  - for once, I'd like the node to crash completely in case it isn't able to sync for 50 days
  - I'd like synching or some other json rpc to inform me that the node has become unsynched with the network

#### Steps to reproduce the behaviour

- no idea; Run erigon for a super long time?

#### Backtrace

````
[backtrace]
````

**Comment by Anish-Agnihotri** [comment]
Any luck figuring this out—seem to have stopped sync at `0xe5a51a` on an old Erigon instance too (probably >1y uptime at this point)?

**Comment by TimDaub** [comment]
Here's what I did:

- Erigon main git repository: Checked out to https://github.com/ledgerwatch/erigon/releases/tag/v2022.08.02
- Then, you'll find that some git submodules also need updates. Update them too
- For libmdbx you'll have to switch the remote https://gitflic.ru/project/erthink/libmdbx/ (context can be found on Twitter. Apparently GH is once again YOLO-deleting state-enemy code America fuck yeah)
- You'll have to remake erigon and rpcdaemon
- Once I restarted erigon, it started rebuilding the database, and this took a while (maybe a day or so)
- Now, everything seemed to have gone back to normal. My node has been running and syncing non stop since I've done these changes
- Sorry for the anti America snark but I am a bit cynical about this

Will close because this issue is essentially resolved for me.
